File: Composite/JavaExample/CompositeMenu/src/Dish.java
public class Dish extends MenuComponent{
    private double price;

    public Dish(String name, double price) {
        super(name);
        this.price = price;
    }

    @Override
    public void show(int indent) {
        String spacing = "  ".repeat(indent);
        System.out.println(spacing + "- " + name + " (" + price + " euro)");
    }
}

File: Composite/CompositeFiles/CompositeFiles/Leaf/Dish.cs
using CompositeFiles.Component;

namespace CompositeFiles.Leaf
{
    public class Dish : MenuComponent
    {
        private double Price;

        public Dish(string name, double price) : base(name)
        {
            this.Price = price;
        }

        public override void Show()
        {
            Console.WriteLine($"{Name} - {Price} € euro.");
        }
    }
}
